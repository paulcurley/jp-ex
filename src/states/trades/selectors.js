export const getAllTrades = (state = { trades: [] }) => state.trades.reverse();
