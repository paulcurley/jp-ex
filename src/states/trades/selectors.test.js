import { getAllTrades } from './selectors';

describe('trade Selector', () => {
    const state = {
        trades: [
            {
                stockSymbol: 'ALC',
                price: 43543,
                noOfShares: 543,
                timestamp: +new Date(),
            },
            {
                stockSymbol: 'VLC',
                price: 888,
                noOfShares: 342,
                timestamp: +new Date(),
            },
        ],
    };
    it('getAllTrades', () => {
        expect(getAllTrades(state)).toMatchSnapshot();
    });
});
