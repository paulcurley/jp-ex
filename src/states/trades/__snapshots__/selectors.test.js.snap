// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trade Selector getAllTrades 1`] = `
Array [
  Object {
    "noOfShares": 342,
    "price": 888,
    "stockSymbol": "VLC",
    "timestamp": 1538657799125,
  },
  Object {
    "noOfShares": 543,
    "price": 43543,
    "stockSymbol": "ALC",
    "timestamp": 1538657799125,
  },
]
`;
